### PR TITLE
added tests for test-wsman

### PR DESCRIPTION
**Pull request by SteveL-MSFT**
Part of https://github.com/PowerShell/PowerShell/issues/4158

### CODE REVIEW

File: test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1 [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+        @{parameter="authentication";value="Negotiate"},
+        @{parameter="authentication";value="ClientCertificate"},
+        @{parameter="authentication";value="Default"},
+        @{parameter="authentication";value="Kerberos"},
+        @{parameter="authentication";value="None"},
+        @{parameter="port";value=5985},
+        @{parameter="port";value=8888},
+        @{parameter="usessl";value=$true},
+        @{parameter="usessl";value=$false},
+        @{parameter="applicationname";value="foo"}
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1]: Could you please indent `value`?

File: test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1 [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+        param($parameter, $value, $expected)
+        $testWsman.$parameter = $value
+        if ($expected -eq $null) {
+            $expected = $value
+        }
+        $testWsman.$parameter | Should Be $expected
+    }
+
+    It "-Authentication for unsupported type should return error" {
+        { Test-WSMan -Authentication foo } | ShouldBeErrorId "CannotConvertArgumentNoMessage,Microsoft.WSMan.Management.TestWSManCommand"
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1]: Please add `-ErrorAction Stop`.

File: test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1 [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    ) {
+        param($computername)
+        $response = Test-WSMan -ComputerName $computername
+        $response.PSObject.TypeNames[0] | Should Be "System.Xml.XmlElement#http://schemas.dmtf.org/wbem/wsman/identity/1/wsmanidentity.xsd#IdentifyResponse"
+        $response.wsmid | Should Be "http://schemas.dmtf.org/wbem/wsman/identity/1/wsmanidentity.xsd"
+        $response.ProtocolVersion | Should Be "http://schemas.dmtf.org/wbem/wsman/1/wsman.xsd"
+        $response.ProductVendor | Should Be "Microsoft Corporation"
+        $response.ProductVersion | Should Be "OS: 0.0.0 SP: 0.0 Stack: 3.0"
+    }
+}
```
Comment (iSazonov) [inline test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1]: Please add new line at EOF.

File: test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1 (hunk 1 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1]: will reformat all the test cases

File: test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1 (hunk 3 above)
Comment (SteveL-MSFT) [reply test/powershell/Modules/Microsoft.WSMan.Management/TestWSMan.Tests.ps1]: yup

**Review by iSazonov — APPROVED** [review]

**Pull request merged**